steward/src/cellars/cellar_v2.rs

```diff
@@ -0,0 +1,702 @@
+//! Handlers for V2 of the Cellar.sol contract functions
+//!
+//! To learn more see https://github.com/PeggyJV/cellar-contracts/blob/main/src/base/Cellar.sol
+use abscissa_core::tracing::debug;
+use ethers::{
+    abi::{self, AbiEncode, Token},
+    contract::EthCall,
+    types::{Address as EthereumAddress, Bytes},
+};
+use steward_abi::cellar_v2::AdaptorCall as AbiAdaptorCall;
+use steward_abi::{
+    aave_a_token_adaptor::AaveATokenAdaptorCalls,
+    aave_debt_token_adaptor::AaveDebtTokenAdaptorCalls, cellar_v2::*,
+    compound_c_token_adaptor::CompoundCTokenAdaptorCalls,
+    uniswap_v3_adaptor::UniswapV3AdaptorCalls, vesting_simple_adaptor::VestingSimpleAdaptorCalls,
+};
+use steward_proto::steward::{
+    aave_a_token_adaptor, aave_debt_token_adaptor,
+    cellar_v2::{adaptor_call::CallData, AdaptorCall, Function as StrategyFunction},
+    compound_c_token_adaptor,
+    oracle_swap_params::Params::{
+        Univ2Params as UniV2OracleParams, Univ3Params as UniV3OracleParams,
+    },
+    swap_params::Params::*,
+    uniswap_v3_adaptor, vesting_simple_adaptor, OracleSwapParams, SwapParams,
+};
+use StrategyFunction::*;
+
+use crate::{
+    error::{Error, ErrorKind},
+    utils::{
+        convert_exchange, sp_call_error, sp_call_parse_address, string_to_u128, string_to_u256,
+    },
+};
+
+use super::log_cellar_call;
+
+const CELLAR_NAME: &str = "CellarV2";
+
+pub fn get_encoded_call(function: StrategyFunction, cellar_id: String) -> Result<Vec<u8>, Error> {
+    match function {
+        AddPosition(params) => {
+            log_cellar_call(CELLAR_NAME, &AddPositionCall::function_name(), &cellar_id);
+            let call = AddPositionCall {
+                index: params.index,
+                position_id: params.position_id,
+                configuration_data: params.configuration_data.into(),
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2Calls::AddPosition(call).encode())
+        }
+        CallOnAdaptor(params) => {
+            log_cellar_call(CELLAR_NAME, &CallOnAdaptorCall::function_name(), &cellar_id);
+            let call = CallOnAdaptorCall {
+                data: get_encoded_adaptor_call(params.data)?,
```

Comment:
Not sure if this is something steward should concern itself with, but is there any upper limit on how much data can be passed into a contract function call? Seems like this could get arbitrarily large.

Reply:
Annoyingly hard to find an answer to. Can only find info about max gas.